src/util_gen.cpp

```diff
@@ -470,6 +477,14 @@ bool Util::isValidFd(int fd)
   return _real_fcntl(fd, F_GETFL, 0) != -1;
 }
 
+// On failure, returns 0, with errno == EFAULT
+bool Util::isReadableMemoryAddress(const char *addr) {
+  // FIXME: We wanted to call libc:open(), but _real_open goes to second wrapper
+  int fd = _real_syscall(SYS_open, "/dev/null", O_WRONLY);
+  int rc = write(fd, addr, 1);
```

Comment:
I did forget to close the fd.  Thanks for pointing that out.  I'll fix that in the next iteration.

As for a write of length 0, I'm worried that some kernels will notice that the write is length 0, and will return success (rc == 0) without testing whether addr is a valid memory address for reading.  Given that we have eventual goals of supporting Android, MacOSX, and just maybe Windows, this seems dangerous.